src/SPC/builder/Extension.php

```diff
@@ -384,20 +407,23 @@ public function buildShared(): void
      */
     public function buildUnixShared(): void
     {
-        $config = (new SPCConfigUtil($this->builder))->config([$this->getName()], with_dependencies: true);
-        [$staticLibString, $sharedLibString] = $this->getStaticAndSharedLibs();
-
-        // macOS ld64 doesn't understand these, while Linux and BSD do
-        // use them to make sure that all symbols are picked up, even if a library has already been visited before
-        $preStatic = PHP_OS_FAMILY !== 'Darwin' ? '-Wl,-Bstatic -Wl,--start-group ' : '';
-        $postStatic = PHP_OS_FAMILY !== 'Darwin' ? ' -Wl,--end-group -Wl,-Bdynamic ' : ' ';
+        $config = (new SPCConfigUtil($this->builder))->config(
+            [$this->getName()],
+            array_map(fn ($l) => $l->getName(), $this->getLibraryDependencies(recursive: true)),
+            $this->builder->getOption('with-suggested-exts'),
+            $this->builder->getOption('with-suggested-libs'),
+        );
+        [$staticLibs, $sharedLibs] = $this->splitLibsIntoStaticAndShared($config['libs']);
         $env = [
             'CFLAGS' => $config['cflags'],
             'CXXFLAGS' => $config['cflags'],
             'LDFLAGS' => $config['ldflags'],
-            'LIBS' => $preStatic . $staticLibString . $postStatic . $sharedLibString,
+            'LIBS' => "-Wl,--start-group {$staticLibs} -Wl,--end-group {$sharedLibs}",
```

Comment:
My local build on mac for shared extension failed with error:

```
ld: unknown options: --start-group --end-group
```

And using previous empty decoration is working well.

Reply:
So far so good for the rest part and my local build. Remember bump version to 2.7.0 too.